websocket/jetty10/src/main/java/jenkins/websocket/Jetty10Provider.java

```diff
@@ -44,21 +45,41 @@
 @MetaInfServices(Provider.class)
 public class Jetty10Provider implements Provider {
 
+    /**
+     * Number of seconds a WebsocketConnection may stay idle until it expires.
+     * Zero to disable.
+     * This value must be higher than the <code>jenkins.websocket.pingInterval</code>.
+     * Per <a href=https://www.eclipse.org/jetty/documentation/jetty-10/programming-guide/index.html#pg-websocket-session-ping>Jetty 10 documentation</a>
+     * a ping mechanism should keep the websocket active. Therefore, the idle timeout must be higher than the ping
+     * interval to avoid timeout issues.
+     */
+    private static long IDLE_TIMEOUT_SECONDS = Long.getLong("jenkins.websocket.idleTimeout", 60L);
```

Comment:
ideally we should have here a default value linked to Long.getLong("jenkins.websocket.pingInterval").
maybe something such `private static long IDLE_TIMEOUT_SECONDS = Long.getLong("jenkins.websocket.idleTimeout", Long.getLong("jenkins.websocket.pingInterval", 30L) * 2);` 
? 
but I agree maybe sounds a bit complicated at least we have now a better default value.

Reply:
I agree with this. They better be correlated. Though changing the default would require to change a value in 2 locations. We already read the property in `WebSocketSession` in `core/`.

I looked into this actually but and it is not straighforward to do it without replicating a `Long.getLong("jenkins.websocket.pingInterval", 30L)` like you suggest here. The jetty implementation is abstracted from core. Core is not accessible from `websocket/`. I thought about adding a method to the `Provider` interface but it looks like the `IdleTimeout` is something specific to Jetty, and not necessarily to other implementations of websocket..

Or maybe we could have `public static` constants in the websocket/spi to share this information.

Reply:
Ideally it should use `jenkins.util.SystemProperties`

Reply:
> it should use `jenkins.util.SystemProperties`

It cannot do so from this module.

Reply:
Linking default values seems too complicated. In the unlikely case you find a need to customize these values, it is not a burden to set two properties.